internal/mail/message.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package mail
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"mime/quotedprintable"
+	"net/mail"
+	"net/textproto"
+	"time"
+)
+
+// Message is RFC compliant email message
+type Message struct {
+	From      mail.Address
+	To        []mail.Address
+	Subject   string
+	ID        string
+	Date      time.Time
+	ReceiptTo []string
+
+	PlainText string
+	Parts     []Part
+}
+
+// Part represent one part of multipart message
+type Part struct {
+	Type        string
+	Encoding    string
+	Disposition string
+	Content     string
+}
+
+// Bytes builds message and returns result as bytes
+func (msg *Message) Bytes() []byte {
+	// always returns nil on read and write, so most of the errors can be ignored
+	var body bytes.Buffer
+
+	// write headers
+	fmt.Fprintf(&body, "MIME-Version: 1.0\r\n")
+	fmt.Fprintf(&body, "Subject: <%v>\r\n", msg.Subject)
```

Comment:
What happens if subject is "\n\n\n\n\n\n"?

Reply:
it breaks everything

Reply:
Use `mime.QEncoding.Encode(msg.Subject)` for the values... if there are multiple values you can pass in `mime.QEncoding.Encode("a", "b", "c")`.